src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -988,9 +989,25 @@ private void DecodeInterceptsLocationAttribute(DecodeWellKnownAttributeArguments
             }
 
             var syntaxTrees = DeclaringCompilation.SyntaxTrees;
-            // PROTOTYPE(ic): consider avoiding an array allocation here, on the assumption that 1 matching tree is the success (common) case, 0 is the most common error case, and 2 or more is much more rare.
-            var matchingTrees = syntaxTrees.WhereAsArray(static (tree, filePath) => tree.FilePath == filePath, filePath);
-            if (matchingTrees is [])
+            SyntaxTree? matchingTree = null;
+            foreach (var tree in syntaxTrees)
```

Comment:
>foreach (var tree in syntaxTrees) [](http://example.com/codeflow?start=12&length=33)

It seems expensive to iterate through the `SyntaxTrees` collection once for each `InterceptsLocationAttribute`. Can we avoid this without too much effort? 

For instance, it looks like we're already building a `HashSet<string>` of file paths in `CSharpCompilation.CheckDuplicateFilePaths()`, at least temporarily. Could we create a `Dictionary<string, SyntaxTree>` instead, and share between this use and `CheckDuplicateFilePaths()`?

Reply:
Makes total sense. We are also gonna have to make a change here to resolve the FilePath before comparing (i.e. handle /pathmap). Do you mind if I leave a comment for it and resolve in future.